Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -213,6 +191,74 @@ public static string GetLanguageVersion(ModuleDefinition module, string targetFr
             }
         }
 
+        // Knowledge on unique machine Ids for different OSes
+        // obtained from https://github.com/denisbrodbeck/machineid
+        public static string GetAnonymizedUserId()
+        {
+            var id = string.Empty;
+            var currentOs = Environment.OSVersion.Platform;
+
+            try
+            {
+                if (currentOs == PlatformID.Win32S || currentOs == PlatformID.Win32Windows ||
+                    currentOs == PlatformID.Win32NT || currentOs == PlatformID.WinCE)
+                {
+                    var rk = Registry.LocalMachine;
+                    id = (string)rk.OpenSubKey("SOFTWARE\\Microsoft\\Cryptography").GetValue("MachineGuid");
+                }
+                else if (currentOs == PlatformID.MacOSX)
+                {
+                    var machineIdToParse = RunProcess("ioreg", "-rd1 -c IOPlatformExpertDevice");
+                    var regex = new Regex("^.*IOPlatformUUID\\\"\\s=\\s\\\"(.+)\\\"", RegexOptions.Multiline);
+                    var match = regex.Match(machineIdToParse);
+                    if (match.Groups.Count == 1)
+                    {
+                        id = match.Groups[0].Value;
+                    }
+                }
+                else if (currentOs == PlatformID.Unix)
+                {
+                    // Some systems only know the /etc path. Sometimes it's the other way round.
+                    string[] linuxIdPaths = new string[] { "/var/lib/dbus/machine-id", "/etc/machine-id" };
+
+                    foreach (var path in linuxIdPaths)
+                    {
+                        try
+                        {
+                            id = File.ReadAllText(path);
+                        }
+                        catch
+                        {
+                            id = string.Empty;
+                        }
+
+                        if (id.Length > 0)
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                if (id.Length == 0)
+                {
+                    return "UNKNOWN";
+                }
+                else
+                {
+                    var salt = new byte[] { 5, 67, 101, 45, 99, 239, 51, 111, 205, 174, 76, 16, 85, 158, 29, 8 };
```

Comment:
If salt is used in all SDKs, then we must be aligned. This is just a proof of concept.